### PR TITLE
UI - add delete for the various kmip models

### PR DESCRIPTION
**Pull request by meirish**
Scopes and roles may now be deleted from the UI and credentials can be revoked.

### CODE REVIEW

File: ui/app/adapters/kmip/role.js [hunk 1]
```diff
@@ -15,19 +17,47 @@ export default BaseAdapter.extend({
...
+    for (let field in nonOperationFields) {
+      if (nonOperationFields[field] == null) {
+        delete nonOperationFields[field];
+      }
+    }
     if (json.operation_all) {
-      return { operation_all: true };
+      return {
+        operation_all: true,
+        ...nonOperationFields,
```
Comment (noelledaley) [inline ui/app/adapters/kmip/role.js]: why do we need to include the `nonOperationFields` now?

File: ui/lib/core/addon/components/menu-loader.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/**
+ * @module MenuLoader
+ * MenuLoader components are used to...
```
Comment (noelledaley) [inline ui/lib/core/addon/components/menu-loader.js]: this description should probably be updated even if it doesn't go into storybook right away. :)

File: ui/app/adapters/kmip/role.js (hunk 1 above)
Comment (madalynrose) [reply ui/app/adapters/kmip/role.js]: we still need to send other things dynamically grabbed from OpenAPI (not all of them are operations)

File: ui/lib/core/addon/components/model-wrap.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { inject as service } from '@ember/service';
```
Comment (madalynrose) [inline ui/lib/core/addon/components/model-wrap.js]: a comment would be good here explaining that this is a wrapper so we can pass methods to be called in

File: ui/app/adapters/kmip/role.js (hunk 1 above)
Comment (meirish) [reply ui/app/adapters/kmip/role.js]: Yes to what @madalynrose, there were other fields added to the endpoint since I first PRd this.

**Pull request merged**